src/tests/AccountCard.test.js

```diff
@@ -17,10 +20,74 @@ const props = {
 }
 
 describe('AccountCard test', () => {
-    it('Render the AccountCard component with the specified props', () => {
-        const component = render(<AccountCard item={props}/>);
+    const deleteButton = jest.fn();
+    const acceptDelete = jest.fn();
+    const cancelDelete = jest.fn();
+
+    it('Render the AccountCard component with the specified props for User', () => {
+        const component = render(<AccountCard item={props} admin={false}/>);
+        expect(component).toBeDefined();
+
+        verifyCardContent(component);
+
+        const button = component.queryByTestId(/btn-delete/i);
+        expect(button).toBeFalsy();
+    });
+
+    it('Render the AccountCard component with the specified props for Admin', () => {
+        const component = render(<AccountCard item={props} admin={true}/>);
         expect(component).toBeDefined();
 
+        verifyCardContent(component);
+
+        expect(component.getByRole('button', {name: /delete/i })).toBeTruthy();
+    });
+
+    it('The account should be deleted when selecting delete', async () => {
+        const component = render(<AccountCard item={props} admin={true}/>);
+        
+        const accountName = component.getByText(props.name);
+        expect(accountName).toHaveTextContent(props.name);
+
+        const typeName = component.getByText(/Standard Checking/i);
+        expect(typeName).toHaveTextContent(props.accountType.name);
+
+        const typeInterestRate = component.getByText(/Interest Rate: 0%/i);
+        expect(typeInterestRate).toHaveTextContent(props.accountType.interestRate);
+
+        const ownershipName = component.getByText(/Ownership: Joint/i);
+        expect(ownershipName).toHaveTextContent(props.ownershipType.name);
+
+        const accountNumber = component.getByText(/3476265/i);
+        expect(accountNumber).toHaveTextContent(props.accountNumber);
+
+        const currentBalance = component.getByText(/720/i);
+        expect(currentBalance).toHaveTextContent(props.currentBalance);
+
+        user.click(screen.getByRole('button', {name: /delete/i }));
+        
+        const modal = component.getByText(/delete account/i);
+        expect(modal).toHaveTextContent('Delete Account');
+
+        user.click(screen.getByRole('button', {name: /save changes/i }));
+
+        waitFor(() => {
+            expect(deleteButton).toHaveBeenCalledTimes(1);
+            expect(acceptDelete).toHaveBeenCalledTimes(1);
+            expect(cancelDelete).toHaveBeenCalledTimes(0);
+        });
+
+        expect(accountName).not.toBeInTheDocument;
+        expect(typeName).not.toBeInTheDocument;
+        expect(typeInterestRate).not.toBeInTheDocument;
+        expect(ownershipName).not.toBeInTheDocument;
+        expect(accountNumber).not.toBeInTheDocument;
+        expect(currentBalance).not.toBeInTheDocument;
```

Comment:
It's not necessary to add all these assertions. If the account name isn't there, the rest of the account data shouldn't be there.